### PR TITLE
support tsconfig `paths`

### PR DESCRIPTION
**Pull request by tachibana-shin**
This pull and https://github.com/esbuild-kit/esm-loader/pull/11 resolve https://github.com/esbuild-kit/tsx/issues/3

### CODE REVIEW

**Comment by morgs32** [comment]
@tachibana-shin and @antfu could we get this one in soon?

**Comment by morgs32** [comment]
Sorry, let me rephrase. I would volunteer to fork this PR and resolve conflicts if that is helpful.

**Comment by tachibana-shin** [comment]
> Sorry, let me rephrase. I would volunteer to fork this PR and resolve conflicts if that is helpful.

Hiroki-san (@privatenumber ) said he wasn't ready to support this feature. When he's ready to support it, I'll work with him to resolve the conflict.

In the meantime that happens you can use the loaders that I patched [esm-loader-fix](https://npmjs.org/package/esm-loader-fix) and [cjs-loader-fix](https://npmjs.org/package/esm-loader-fix) used in [epact framework](https://github.com/tachibana-shin/epact)

**Comment by privatenumber** [comment]
Thanks for your patience

I decided to move forward with tsconfig `paths` using this implementation: https://github.com/esbuild-kit/cjs-loader/pull/8

 [Reason explained here](https://github.com/esbuild-kit/esm-loader/pull/11#issuecomment-1150565372)
